tests/raftstore/test_conf_change.rs

```diff
@@ -429,3 +431,73 @@ fn test_server_after_remove_itself() {
     let mut cluster = new_server_cluster(0, count);
     test_after_remove_itself(&mut cluster);
 }
+
+fn test_split_brain<T: Simulator>(cluster: &mut Cluster<T>) {
+    let r1 = cluster.bootstrap_conf_change();
+    cluster.start();
+    cluster.change_peer(r1, ConfChangeType::AddNode, new_peer(2, 2));
+    cluster.change_peer(r1, ConfChangeType::AddNode, new_peer(3, 3));
+
+    // leader isolation
+    let leader = cluster.leader_of_region(r1);
+    assert_eq!(leader, Some(new_peer(1, 1)));
+    cluster.hook_transport(Isolate::new(1));
+
+    // refresh region info, maybe no need
+    cluster.must_put(b"k1", b"v1");
+
+    // add [4,5,6] and remove [2,3]
+    cluster.change_peer(r1, ConfChangeType::AddNode, new_peer(4, 4));
+    cluster.change_peer(r1, ConfChangeType::AddNode, new_peer(5, 5));
+    cluster.change_peer(r1, ConfChangeType::AddNode, new_peer(6, 6));
+    cluster.change_peer(r1, ConfChangeType::RemoveNode, new_peer(2, 2));
+    cluster.change_peer(r1, ConfChangeType::RemoveNode, new_peer(3, 3));
+
+    // refresh region info, maybe no need
+    cluster.must_put(b"k2", b"v2");
+
+    // when network recovers, 1 will send request vote to [2,3]
+    cluster.reset_transport_hooks();
+    cluster.partition(vec![1, 2, 3], vec![4, 5, 6]);
+
+    // refresh region info, maybe no need
+    cluster.must_put(b"k3", b"v3");
+
+    // check whether a new cluster [1,2,3] is formed
+    // if so, both [1,2,3] and [4,5,6] think they serve for region r1
+    // result in split brain
+    let header0 = find_leader_response_header(cluster, r1, new_peer(2, 2));
+    assert!(header0.get_error().has_region_not_found());
+
+    // at least wait for a round of election timeout and check again
+    let term1 = find_leader_response_header(cluster, r1, new_peer(1, 1)).get_current_term();
+    sleep_ms(500);
+    let term2 = find_leader_response_header(cluster, r1, new_peer(1, 1)).get_current_term();
+    assert!(term2 > term1);
```

Comment:
not always, but it may fail even on my own machine if sleep for only 500ms.
I haven't seen panic locally if sleep for 1 seconds, but I suggest reset to pervious version.